Add parquet support

zhkvia opened the pull request:
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

zhkvia committed:
Add a parquet-cpp repo to contrib

zhkvia committed:
Add cmake modules for libraries detection

zhkvia committed:
Link parquet and arrow libraries

zhkvia committed:
Add a template for ParquetBlockInputStream

zhkvia committed:
Parquet input: support numeric types

zhkvia committed:
Parquet input: support a string type

ludv1x commented on dbms/src/DataStreams/ParquetBlockInputStream.cpp:
```diff
@@ -0,0 +1,169 @@
+    if (istr.eof())
+        return res;
+
+    // TODO: rewrite this garbage
```
Use `void copyData(ReadBuffer & from, WriteBuffer & to);` where to is `WriteBufferFromString`

ludv1x commented on dbms/src/DataStreams/ParquetBlockInputStream.cpp:
```diff
@@ -0,0 +1,169 @@
+    // TODO: Use an internal Exception?
+    PARQUET_THROW_NOT_OK(filereader.ReadTable(&table));
+
+    if (0 == table->num_rows())
```
I am not sure that it is an error.

zhkvia committed:
Parquet input: get rid of redundant copying

zhkvia committed:
Parquet input: add a boolean type support

zhkvia committed:
Parquet input: change if to switch

zhkvia committed:
Parquet input: add a Date type support

zhkvia committed:
Parquet output: support numeric and string types

zhkvia committed:
Check arrow status

zhkvia committed:
Parquet output: refactor functions-fillers

zhkvia committed:
Actualize TODOs

zhkvia committed:
Parquet input: support NULL values

zhkvia committed:
Refactor: make some members and methods static

zhkvia committed:
Update submodules since parquet is a part of arrow

- https://github.com/apache/parquet-cpp/commit/f84dca4203930fa2898d0d98d278d74520fafec4

zhkvia committed:
Parquet output: support NULL values

zhkvia committed:
Parquet data format: refactoring

zhkvia committed:
Parquet data format: update CMake configurations
